File: pr42862.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    vector<int> student(n, 1);
        for (int j = 0; j < lost.size(); j++) {
			if (student[lost[j] - 1] != 0)
				student[lost[j] - 1]--;
		}
		for (int j = 0; j < reserve.size(); j++) {
			student[reserve[j] - 1] += 1;
		}
		

		for (int i = 0; i < n; i++) {
			if (student[i] == 0) {
				if (student[i - 1] == 2 && i > 0) {
					student[i - 1]--;
					student[i]++;
				}
				else if (student[i + 1] == 2 && i + 1 < n) {
					student[i + 1]--;
					student[i]++;
				}
			}
		}
		int count = 0;
		for (int i = 0; i < n; i++) {
			if (student[i] != 0)
				count++;
		}

		cout << count;

	return count;
}

File: pr42862.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        int[] st = new int[n];
        //st배열 1로 초기화
        for(int i = 0; i < st.length; i++){
            st[i] = 1;
        }
        //reserve면 +1, lost면 -1
        for(int i = 0; i < st.length; i++){
            for(int j = 0; j < reserve.length; j++){
                if(reserve[j] == i+1)
                    st[i]++;
            }
            for(int j = 0; j < lost.length; j++){
                if(lost[j] == i+1)
                    st[i]--;
            }
        }
        for(int i = 0; i < st.length; i++){
            //st[i] == 0 이면
            if(st[i] == 0){
                if(i > 0 && st[i-1]==2){
                    st[i-1]--;
                    st[i]++;
                }
                else if(i+1 < n && st[i+1]==2){
                    st[i+1]--;
                    st[i]++;
                }
                else continue;
            }
        }
        for(int i = 0; i < st.length; i++){
            if(st[i] != 0)
                answer++;
        }
        
        return answer;
    }
}